src/radio/radio-group.component.ts

```diff
@@ -0,0 +1,407 @@
+	@HostBinding("attr.role") role = "radiogroup";
+
+	/**
+	 * Binds 'bx--radio-button-group' value to the class for `RadioGroup`.
+	 * @memberof RadioGroup
+	 */
+	@HostBinding("class") class = "bx--radio-button-group";
+
+	/**
+	 * To track whether the `RadioGroup` has been initialized.
+	 * @private
+	 * @memberof RadioGroup
+	 */
+	private isInitialized = false;
+	/**
+	 * Reflects whether or not the input is disabled and cannot be selected.
+	 * @type {boolean}
+	 * @private
+	 * @memberof RadioGroup
+	 */
+	private _disabled = false;
+	/**
+	 * The value of the selected option within the `RadioGroup`.
+	 * @private
+	 * @type {any}
+	 * @memberof RadioGroup
+	 */
+	private _value: any = null;
+	/**
+	 * The `RadioComponent` within the `RadioGroup` that is selected.
+	 * @private
+	 * @type {RadioComponent}
+	 * @memberof RadioGroup
+	 */
+	private _selected: RadioComponent = null;
+	/**
+	 * The name attribute associated with the `RadioGroup`.
+	 * @type {string}
+	 * @private
+	 * @memberof RadioGroup
+	 */
+	private _name = `radio-group-${RadioGroup.radioGroupCount}`;
+
+	/**
+	 * Creates an instance of RadioGroup.
+	 * @param {ChangeDetectorRef} changeDetectorRef
+	 * @param {ElementRef} elementRef
+	 * @param {Renderer2} renderer
+	 * @memberof RadioGroup
+	 */
+	constructor(private changeDetectorRef: ChangeDetectorRef, private elementRef: ElementRef, private renderer: Renderer2) {
+		RadioGroup.radioGroupCount++;
+	}
+
+	/**
+	 * Updates the selected `RadioComponent` to be checked (selected).
+	 * @memberof RadioGroup
+	 */
+	checkSelectedRadio() {
+		if (this._selected && !this._selected.checked) {
+			this._selected.checked = true;
+		}
+	}
+
+	/**
+	 * Use the value of the `RadioGroup` to update the selected radio to the right state (selected state).
+	 * @memberof RadioGroup
+	 */
+	updateSelectedRadioFromValue() {
+		let alreadySelected = this._selected != null && this._selected.value === this._value;
+
+		if (this.radios != null && !alreadySelected) {
+			this._selected = null;
+			this.radios.forEach(radio => {
+				radio.checked = this.value === radio.value;
+				if (radio.checked) {
+					this._selected = radio;
+				}
+			});
+		}
+	}
+
+	/**
+	 * Creates a class of `RadioChange` to emit the change in the `RadioGroup`.
+	 * @memberof RadioGroup
+	 */
+	emitChangeEvent() {
+		if (this.isInitialized) {
+			let event = new RadioChange();
+			event.source = this._selected;
+			event.value = this._value;
+			this.change.emit(event);
+		}
+	}
+
+	/**
+	 * Calls the `markForCheck()` function within the `changeDetectorRef` class
+	 * to trigger Angular's change detection on each radio item.
+	 * @memberof RadioGroup
+	 */
+	markRadiosForCheck() {
+		if (this.radios) {
+			this.radios.forEach(radio => radio.markForCheck());
+		}
+	}
+
+	/**
+	 * Synchronizes the names of the radio items with the name of the `RadioGroup`.
+	 * @memberof RadioGroup
+	 */
+	updateRadioNames() {
+		if (this.radios) {
+			this.radios.forEach(radio => {
```

Comment:
This can technically be all on one line, no?

```suggestion
			this.radios.forEach(radio => radio.name = this.name);
```